test/micro/org/openjdk/bench/jdk/incubator/vector/LoadMaskedIOOBEBenchmark.java

```diff
@@ -0,0 +1,141 @@
+// 2 along with this work; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+// or visit www.oracle.com if you need additional information or have any
+// questions.
+//
+//
+package org.openjdk.bench.jdk.incubator.vector;
+
+import java.util.concurrent.TimeUnit;
+import jdk.incubator.vector.*;
+import org.openjdk.jmh.annotations.*;
+
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+@Warmup(iterations = 3, time = 1)
+@Measurement(iterations = 5, time = 1)
+@Fork(value = 1, jvmArgsPrepend = {"--add-modules=jdk.incubator.vector"})
+public class LoadMaskedIOOBEBenchmark {
+    @Param({"1026"})
+    private int inSize;
+
+    @Param({"1152"})
+    private int outSize;
+
+    private byte[] byteIn;
+    private byte[] byteOut;
+    private short[] shortIn;
+    private short[] shortOut;
+    private int[] intIn;
+    private int[] intOut;
+    private long[] longIn;
+    private long[] longOut;
+    private float[] floatIn;
+    private float[] floatOut;
+    private double[] doubleIn;
+    private double[] doubleOut;
+
+    private boolean[] m;
+
+    private static final VectorSpecies<Byte> bspecies = VectorSpecies.ofLargestShape(byte.class);
+    private static final VectorSpecies<Short> sspecies = VectorSpecies.ofLargestShape(short.class);
+    private static final VectorSpecies<Integer> ispecies = VectorSpecies.ofLargestShape(int.class);
+    private static final VectorSpecies<Long> lspecies = VectorSpecies.ofLargestShape(long.class);
+    private static final VectorSpecies<Float> fspecies = VectorSpecies.ofLargestShape(float.class);
+    private static final VectorSpecies<Double> dspecies = VectorSpecies.ofLargestShape(double.class);
+
+    @Setup(Level.Trial)
+    public void Setup() {
+        byteIn = new byte[inSize];
+        byteOut = new byte[outSize];
+        shortIn = new short[inSize];
+        shortOut = new short[outSize];
+        intIn = new int[inSize];
+        intOut = new int[outSize];
+        longIn = new long[inSize];
+        longOut = new long[outSize];
+        floatIn = new float[inSize];
+        floatOut = new float[outSize];
+        doubleIn = new double[inSize];
+        doubleOut = new double[outSize];
+
+        for (int i = 0; i < inSize; i++) {
+            byteIn[i] = (byte) i;
+            shortIn[i] = (short) i;
+            intIn[i] = i;
+            longIn[i] = i;
+            floatIn[i] = (float) i;
+            doubleIn[i] = (double) i;
+        }
+        m = new boolean[outSize];
+        for (int i = 0; i < inSize; i++) {
+            m[i] = i % 2 == 0;
+        }
+    }
+
+    @Benchmark
+    public void byteLoadArrayMaskIOOBE() {
+        for (int i = 0; i < inSize; i += bspecies.length()) {
+            VectorMask<Byte> mask = VectorMask.fromArray(bspecies, m, i);
```

Comment:
For other case "if (offset >= 0 && offset <= (a.length - species.length())) )" we are anyways intrinsifying, should we limit this micro to work only for newly optimized case.

Reply:
Yeah, thanks and it's really a good suggestion to limit this benchmark only for the IOOBE cases. I locally modified the tests to make sure only the IOOBE case happens and the results show good as well. But do you think it's better to keep as it is since we can also see the performance of the common cases to make sure no regressions happen? As the current benchmarks can also show the performance gain by this PR.

Reply:
It was just to remove the noise from a targeted micro benchmark. But we can keep it as it is.

Reply:
OK, thanks!